x-pack/solutions/security/plugins/security_solution/scripts/telemetry/build_ebt_data_view.ts

```diff
@@ -146,40 +146,89 @@ function removeTrailingSlash(url: string) {
     return url;
   }
 }
+interface NestedSchemaNode {
+  type?: string;
+  properties?: NestedObject;
+  items?: NestedSchemaNode;
+  [key: string]: unknown;
+}
 interface NestedObject {
-  [key: string]: { type?: string; properties?: NestedObject };
+  [key: string]: NestedSchemaNode;
+}
+
+interface InspectSchemaNodeResult {
+  childToInspect?: NestedObject;
+  nodeType?: string;
+}
+
+function isObjectRecord(x: unknown): x is Record<string, unknown> {
+  return typeof x === 'object' && x !== null;
+}
+
+function inspectArraySchemaNode(node: NestedSchemaNode): InspectSchemaNodeResult {
+  const item = node.items;
+
+  if (!isObjectRecord(item)) {
+    return { nodeType: 'array' };
+  }
+
+  if ('type' in item && (item as NestedSchemaNode).type) {
+    const t = String((item as NestedSchemaNode).type);
+    if (t === 'array') return { nodeType: 'array' }; // array-of-arrays -> keep "array"
+    return { nodeType: t }; // array of primitives
+  }
+
+  if (
+    (item as NestedSchemaNode).properties &&
+    isObjectRecord((item as NestedSchemaNode).properties)
+  ) {
+    return { childToInspect: (item as NestedSchemaNode).properties as NestedObject }; // array of objects
+  }
+
+  return { nodeType: 'array' };
 }
 
-function flattenSchema(inputObj: NestedObject): { [key: string]: string } {
-  const result: { [key: string]: string } = {};
+function inspectSchemaNode(node: NestedSchemaNode): InspectSchemaNodeResult {
+  if (!node.type) {
+    const objectNode = node.properties ?? (node as unknown as NestedObject);
+    return { childToInspect: objectNode };
+  }
+
+  if (node.type === 'array') {
+    return inspectArraySchemaNode(node);
+  }
+
+  return { nodeType: String(node.type) };
+}
+
+export function flattenSchema(inputObj: NestedObject): Record<string, string> {
+  const result: Record<string, string> = {};
```

Comment:
nit: Let's move these functions to this file's bottom. These inspect functions are implementation details so it's logical to see the exported members first.